Extension/src/LanguageServer/client.ts

```diff
@@ -3745,6 +3735,17 @@ export class DefaultClient implements Client {
     public setReferencesCommandMode(mode: refs.ReferencesCommandMode): void {
         this.model.referencesCommandMode.Value = mode;
     }
+
+    public addTrustedCompiler(path: string): Promise<void> {
+        // Detect duplicate paths or invalid paths.
+        if (trustedCompilerPaths.includes(path) || path === null || path === undefined) {
+            return Promise.resolve();
+        }
+        trustedCompilerPaths.push(path);
+        return this.requestCompiler(path).then((result) => {
```

Comment:
Should this use `await` instead of `then` ?  It seems like this function could be simplified slightly if async.

Reply:
No -- I didn't want to force all the callers to be async.

Reply:
Oh, I was mistaken.